scripts/common.make

```diff
@@ -31,6 +31,27 @@ ifdef WINDOWS
 WINDOWS_BUILD=1
 endif
 
+## Usage: $(call remove_unset_options,PREPROCESSOR_INPUT)
+## Remove the preprocessor symbols that are not set in the current configuration
+## from PREPROCESSOR_INPUT. Also normalize whitespace.
+## Example:
+##   $(call remove_unset_options,MBEDTLS_FOO MBEDTLS_BAR)
+## This expands to an empty string "" if MBEDTLS_FOO and MBEDTLS_BAR are both
+## disabled, to "MBEDTLS_FOO" if MBEDTLS_BAR is enabled but MBEDTLS_FOO is
+## disabled, etc.
```

Comment:
> This expands to an empty string "" if MBEDTLS_FOO and MBEDTLS_BAR are both disabled

I don't think this is correct. This expands to "MBEDTLS_FOO MBEDTLS_BAR" if they are both disabled. It expands to an empty string if they are both enabled.

This should be called "remove_set_options" as it does exactly that. Unset options are left in the list.

Reply:
Indeed. There's a negation compared to what I'd prefer (something truish if the options are set), because it's easier to implement this way. And I somehow included the negation twice in the name and part of the documentation. I've pushed a fix.